velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -207,6 +207,59 @@ struct UnixTimestampParseWithFormatFunction
   bool invalidFormat_{false};
 };
 
+/// Converts date string to Timestmap type.
+template <typename T>
+struct GetTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  int16_t getTimezoneId(const DateTimeResult& result) {
+    // If timezone was not parsed, fallback to the session timezone. If there's
+    // no session timezone, fallback to 0 (GMT).
+    return result.timezoneId != -1 ? result.timezoneId
+                                   : sessionTimezoneId_.value_or(0);
+  }
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*input*/,
+      const arg_type<Varchar>* format) {
+    auto sessionTimezoneName = config.sessionTimezone();
+    if (!sessionTimezoneName.empty()) {
+      sessionTimezoneId_ = util::getTimeZoneID(sessionTimezoneName);
+    }
+    if (format != nullptr) {
+      this->formatter_ = buildJodaDateTimeFormatter(
+          std::string_view(format->data(), format->size()));
+      isConstantTimeFormat_ = true;
```

Comment:
Seems `isConstantTimeFormat_` can be removed by checking if this->formatter_ is nullptr instead.

Reply:
Maybe, better to keep it since formatter_ can be null for some illegal cases.